File: J_se_les12/HW/Core/Logic/Save/SaveBase.java
package Java.J_se_les12.HW.Core.Logic.Save;

import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public abstract class SaveBase {
    Phonebook phonebook;

    protected SaveBase(Phonebook phonebook) {
        this.phonebook = phonebook;
    }

    public abstract void save(String type, String path);

}

File: J_se_les13/HW/Core/Logic/Save/SaveBase.cs
using HW.Core.Phonebook;
namespace HW.Core.Logic.Save
{
    public abstract class SaveBase
    {
        public PhonebookV1 Phonebook { get; set; }
        protected SaveBase(PhonebookV1 phonebook)
        {
            Phonebook = phonebook;
        }
        public abstract void Save(string type, string path);
    }
}